torch/csrc/api/include/torch/data/example.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <torch/tensor.h>
+
+namespace torch {
+namespace data {
+
+/// An `Example` from a dataset.
+///
+/// A dataset consists of data and an associated target (label).
+template <typename Data = Tensor, typename Target = Tensor>
+struct Example {
```

Comment:
It would be nice to write up a toy script that does some something else than standard supervised learning, and see how painful are the template parameters if you need to use a different example type.

Actually it might be enough to try to thread your own custom `Example` type which is equivalent to this one, but is a different type.